### PR TITLE
DDF-3139 Adds reentrant lock to prevent concurrent scheduled and explicit sets from stepping on each other

### PR DESCRIPTION
**Pull request by coyotesqrl**
Also updates test resource CRLs so they expire in 100 years. This will prevent unit tests failing because the CRL reference is being updated after being explicitly set.

#### What does this PR do?
Fixes timing issue impacting unit tests. It is unlikely to ever have impacted production code, but this will remove that slight possibility.

#### Who is reviewing it? 
@rzwiefel @kcwire 

(please choose AT LEAST two reviewers that need to approve the PR before it can get merged)
#### Select relevant component teams: 
@codice/security

#### Choose 2 committers to review/merge the PR. 
(please choose ONLY two committers from below, delete the rest)
@clockard
@stustison

#### How should this be tested? (List steps with links to updated documentation)
Full build/test run should be sufficient.

#### Any background context you want to provide?
N/A

#### What are the relevant tickets?
[DDF-3139](https://codice.atlassian.net/browse/DDF-3139)

#### Screenshots (if appropriate)
N/A

#### Checklist:
- [ ] Documentation Updated
- [X] Update / Add Unit Tests
- [ ] Update / Add Integration Tests
#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java [hunk 1]
```diff
@@ -160,18 +169,23 @@ public void run() {
...
-                LOGGER.info("CRL property in {} is not set. Certs will not be checked against a CRL",
-                        encryptionPropertiesLocation);
-                crlCache.set(null);
-            } else {
-                CRL crl = createCrl(location);
-                if (crl != null) {
-                    crlCache.set(crl);
-                    LOGGER.info("CRL has been updated from {}.", location);
+        void setCrlLocation(String location) {
+            lock.lock();
```
Comment (coyotesqrl) [inline platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java]: @stustison This reentrant lock will prevent concurrent access from the scheduled updater and the explicit call to set the location; however...never mind. I was going to say that anyone explicitly setting the `crlLocation` would have their value replaced on the next scheduled check/next day depending on the `nextUpdate` value of the crl, but I see that the wrapping class's `setCrlLocation` method - is only ever invoked by unit test.

**Comment by coyotesqrl** [comment]
build now

File: platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java [hunk 2]
```diff
@@ -126,31 +128,38 @@ URL urlFromPath(String location) {
...
+                    CRL_PROPERTY_KEY);
+
+            lock.lock();
+            try {
+                CRL crl = crlCache.get();
+                if (crl instanceof X509CRL) {
+                    X509CRL x509Crl = (X509CRL) crl;
+                    if (Calendar.getInstance()
+                            .getTime()
+                            .after(x509Crl.getNextUpdate()) || rolledOverDay()) {
```
Comment (rzwiefel) [inline platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java]: ❗️ NPE here if getNextUpdate returns null

Comment (rzwiefel) [inline platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java]: A possible, null-safe implementation of this method:
```java
        public void run() {
            Properties encryptionProperties = loadProperties(encryptionPropertiesLocation);
            CRL crl = crlCache.get();

            if (crl == null || !(crl instanceof X509CRL)) {
                return;
            }

            boolean isBeforeNextUpdate = Optional.of((X509CRL) crl)
                    .map(X509CRL::getNextUpdate)
                    .map(Date::toInstant)
                    .map(t -> Instant.now().isBefore(t))
                    .orElse(true);
            if (isBeforeNextUpdate && !rolledOverDay()) {
                return;
            }

            setCrlLocation(encryptionProperties.getProperty(CRL_PROPERTY_KEY));
            start = Calendar.getInstance();
        }
```

Comment (coyotesqrl) [inline platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java]: That would be wrong, as it no longer sets the location when it wasn't previously set. I've updated it to handle the NPE on `getNextUpdate()`.

File: platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java [hunk 3]
```diff
@@ -126,31 +128,35 @@ URL urlFromPath(String location) {
...
-                        .getTime()
-                        .after(x509Crl.getNextUpdate()) || rolledOverDay()) {
-                    setCrlLocation(encryptionProperties.getProperty(CRL_PROPERTY_KEY));
-                    start = Calendar.getInstance();
+            String crlLocation = loadProperties(encryptionPropertiesLocation).getProperty(
+                    CRL_PROPERTY_KEY);
+
+            lock.lock();
+            try {
+                CRL crl = crlCache.get();
```
Comment (coyotesqrl) [inline platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java]: Conditional logic cleaned and tightened a bit here.

**Comment by coyotesqrl** [comment]
build now

**Comment by coyotesqrl** [comment]
build now

**Comment by coyotesqrl** [comment]
build now

**Comment by coyotesqrl** [comment]
build now

**Pull request merged**